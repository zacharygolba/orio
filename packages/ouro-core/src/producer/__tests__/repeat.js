// @flow

import Repeat from '../repeat'

const ITEM = 'test'
let producer

beforeEach(() => {
  producer = new Repeat(ITEM)
})

test('#constructor()', () => {
  expect(producer).toMatchSnapshot()
})

test('#@@iterator()', () => {
  let i = 0

  for (const item of producer) {
    if (i > 3) {
      break
    }

    expect(item).toBe(ITEM)
    i += 1
  }
})
