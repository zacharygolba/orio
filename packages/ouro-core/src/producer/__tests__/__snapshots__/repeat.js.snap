// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#constructor() 1`] = `
RepeatProducer {
  "value": "test",
}
`;
